packages/caravan-wallets/src/trezor.ts

```diff
@@ -602,6 +614,30 @@ export class TrezorExportPublicKey extends TrezorExportHDNode {
     }
     return payload.publicKey;
   }
+
+  /** 
+   * Helper to parse BIP-380 descriptor string into fingerprint + xpub.
+   * Example descriptor: pkh([d34db33f/48h/0h/0h/2h]xpub6CUGRU.../0/*) 
+   */ 
+  parseDescriptor(descriptor: string) { 
+    let rootFingerprint = ""; let xpub = ""; 
+    // Extract fingerprint 
+    // (inside square brackets before the first slash) 
+    const fpMatch = descriptor.match(/ \[([0-9a-fA-F]+)\//); 
+    if (fpMatch) { rootFingerprint = fpMatch[1]; } 
+    
+    // Extract xpub (the base58 string starting with "xpub" or "tpub") 
+    const xpubMatch = descriptor.match(/(xpub[0-9A-Za-z]+|tpub[0-9A-Za-z]+)/); 
```

Comment:
we have xpub parsing in @caravan/bitcoin in the keys.ts file that we can use to have more robust checking.